doc/release-process.md

```diff
@@ -1,14 +1,54 @@
 # Release Process
 
-1. Open PR to master that
-   1. adds release notes to `doc/CHANGELOG.md` and
-   2. if this is **not** a patch release, updates `_PKG_VERSION_{MAJOR,MINOR}` and `_LIB_VERSIONS_*` in `configure.ac`
-2. After the PR is merged,
-   * if this is **not** a patch release, create a release branch with name `MAJOR.MINOR`.
-     Make sure that the branch contains the right commits.
-     Create commit on the release branch that sets `_PKG_VERSION_IS_RELEASE` in `configure.ac` to `true`.
-   * if this **is** a patch release, open a pull request with the bugfixes to the `MAJOR.MINOR` branch.
-     Also include the release note commit bump `_PKG_VERSION_BUILD` and `_LIB_VERSIONS_*` in `configure.ac`.
-4. Tag the commit with `git tag -s vMAJOR.MINOR.PATCH`.
-5. Push branch and tag with `git push origin --tags`.
-6. Create a new GitHub release with a link to the corresponding entry in `doc/CHANGELOG.md`.
+This document outlines the process for releasing versions of the form `$MAJOR.$MINOR.$PATCH`.
+
+We distinguish between two types of releases: *regular* and *maintenance* releases.
+Regular releases are releases of a new major or minor version as well as patches of the most recent release.
+Maintenance releases, on the other hand, are required for patches of older releases.
+
+Note that adding a release entry to [CHANGELOG.md](../CHANGELOG.md) requires a release date. This date should match the dates of the remaining steps in the release process, including the date of the tag and the GitHub release.
+
+This process also assumes that there will be no minor releases for old major releases.
+
+## Regular release
+
+1. Open PR to the master branch that includes the following commits:
```

Comment:
Urghs, this is a good point. (And this was not a problem with the previous approach...)

> Don't care about it

 Well we anyway need to open the PR for the right date to get the release date right... So yeah, this is doable but it still seems error-prone. (And releases are the one thing where we don't want to make mistakes, because we cannot correct them without another release.)

> Use two separate PRs, and tag the merge-commit of the first.

Ok, that works. I tried to avoid two PRs but ok.  What if we have a single PR, merge that locally with the merge tool, tag the merge, let the merging maintainer add the "bump after release" commit manually, and then push the result. That saves the second PR and having that one commit with PR is harmless. (And if there's a problem with the "bump after release" commit, then we can just fix it on master and it won't affect the release.) 

> Use two commits (or at least the first one) that's not part of a PR, but just immediately pushed by a maintainer to the master branch.

Is that what I just wrote above? I'm not sure. You say at least the first commit should be manual and not part of the PR, and I don't see how that would work. (In my suggestion, the *second* commit is done manually.)

Reply:
> I don't see why the second commit matters? It's past the tag, so it can't affect it.
> 
> All that matters is that there is a stable point on the master branch after the first commit. That can be achieved by it being in a separate PR, and using the merge commit of that PR as the tag, or by just pushing a commit and marking that as the tag.

Yes, we're in full agreement here! 

So I think we should
1. tag the merge commit of a PR
2. then get the "release: update master ..." commit on master (with or without a PR). 

Now that I think about  this again, I tend so say let's make a PR so that we never break the implicit "don't push without PR" rule (this also means other maintainers will see it etc.) But to make sure the process is not overly complicated and interactive, I suggest that the maintainer preparing the release simply merges this second PR without an ACK. (And indeed, even if we then spot a mistake in the second PR, it won't affect the release).

Reply:
That's quite unfortunate, I did not consider this.

> So I think we should
> 1. tag the merge commit of a PR
> 2. then get the "release: update master ..." commit on master (with or without a PR).

One issue with this approach is that `IS_RELEASE` is set to true in the merge commit as well as in the commit that set it to true. Therefore, we have two (maybe significantly) different states of the repo that are marked as `IS_RELEASE`. Another problem with this approach is that the repo is locked after the first PR until the second commit lands in the repo (unless you ignore `IS_RELEASE` and `VERSION_PATCH` for a while).

Another approach is that we revert the release process back to a version where we set the patch version of master to `99` and branch off from before we set `IS_RELEASE` which is preserved [here](https://github.com/jonasnick/secp256k1/blob/initial-release-backup2/doc/release-process.md). This version, however, has some subtle edge cases that made me prefer the current version of the release process.

I would have no problem with the approach that locks the repo until the PR is merged ("don't care about it") if we could reasonably ensure that no mistakes will happen. But I have no better ideas than mentioning it in the release process and in the OP of the PR.

Reply:
As per our discussion elsewhere I changed the release process to @real-or-random's suggestion above.